website/content/community/release-process.md

```diff
@@ -13,6 +13,19 @@ Are you actually ready to release? Check the milestone on github and
 verify that all its issues are closed. If there are open issues,
 you'll have to either resolve them, or bump to the next version.
 
+### Check for outdated dependencies
+
+This might be a little late to do at release time, but releases are not so
+frequent for now, so it makes sense to do this before creating a release.
+
+Some dependencies are usually best to stay fresh, like client-go and friends.
+But ideally we should be using recent versions of all our dependencies.
+
+If client-go is too old (like 2-3 releases old), from time to time Kubernetes
+may remove some field that breaks MetalLB if it is using an old version. Make
+sure either to use a recent version or create an issue to do a new release,
+after proper testing, to update client-go.
```

Comment:
My only concern here is that I don't want to imply we should do big dependency bumps and then release right away.  I think we'd want big jumps to soak for a bit in CI.  I *think* that's what you're saying here by saying to create an issue.

Would you like to be more strict and saying something like "we should not issue a new release with a client-go that is too old." and have that as part of the release checklist and what would potentially make us hold off a release?  It's not clear from the text here if you think a release should be held over this.

Reply:
Another way to have more testing that was not used in the past is doing RC to get more testing from the field.
ie before a major release update dependencies that make sense and build a RC to get feedback.
I don't think many people test/run master ;)

Reply:
yes RCs for the feature releases are a good idea, too.